R/print.R

```diff
@@ -54,10 +54,10 @@ print.dataverse_dataset <- function(x, ...) {
             cat("Release Date: ", x$releaseTime, "\n", sep = "")
         }
         if ("license" %in% names(x)) {
-            cat("License: ", x$license, "\n", sep = "")
+            cat("License: ", x$license$name,"\n", sep = "")
```

Comment:
I don't know when this change occurred ('license' is now a list of name / uri / logo uri), or whether one strives for backward compatibility. Using just `x$license` creates an error and the remainder of the print method is not

Reply:
Probably this change came with multiple license support in Dataverse 5.10: https://github.com/IQSS/dataverse/releases/tag/v5.10

Reply:
thanks; that's quite a while ago (2022?). Is there an example of a dataset with multiple licenses, or a pointer to where I can find the structure returned by <https://guides.dataverse.org/en/latest/api/native-api.html#get-version-of-a-dataset> which I think is the endpoint used to retrieve dataset metadata in the R code. For a single license it seems like the return has a field `license` which is a JSON object; is the return `license` with a JSON array of objects?

Reply:
Well, a dataset can only have license or "custom" terms.

It looks like you've discovered how a license can look:


    "license": {
      "name": "CC0 1.0",
      "uri": "http://creativecommons.org/publicdomain/zero/1.0",
      "iconUri": "https://licensebuttons.net/p/zero/1.0/88x31.png"
    }

... that's from here, for example: https://dataverse.harvard.edu/api/datasets/:persistentId/versions/:latest?persistentId=doi%3A10.7910/DVN/TJCLKP

Custom terms can look like this:

    "termsOfUse": "<div style=\"padding-left: 30px;\"> <ul style=\"list-style-type: decimal;\" ><li> The Murray Archive (the Distributor) has granted me a revocable license to use this dataset solely for the purposes of conducting research, and the Distributor may terminate this license at any time and for any reason. </li> <li> I will use the dataset solely for statistical analysis and reporting of aggregated information, and not for investigation of specific individuals or organizations, except when identification is authorized in writing by the Distributor.   </li> <li>I will produce no links among the Distributor’s datasets or among the Distributor’s data and other datasets that could identify individuals or organizations. </li> <li>I represent that neither I, nor anyone I know, has any prior knowledge of the possible identities of any study participants in any dataset that I am being licensed to use.   </li> <li> I will not knowingly divulge any information that could be used to identify individual participants in the study, nor will I attempt to identify or contact any study participant, and I agree to use any precautions necessary to prevent such identification.\r\n</li> <li> I will make no use of the identity of any person or establishment discovered inadvertently.  If I suspect that I might recognize or know a study participant, I will immediately inform the Distributor, and I will not use or retain a copy of data regarding that study participant.  If these measures to resolve an identity disclosure are not sufficient, the Distributor may terminate my use of the dataset. </li> <li> I will not reproduce the dataset except as is necessary for my scholarly purposes.  I will destroy the dataset upon the completion of my scholarly work with it.\r\n</li> <li> I will not share data from the dataset (in any form or by any means) with any third party, including other members of my research team, as I understand that all users of data must obtain the data directly from the Distributor. </li> <li> I will make appropriate acknowledgement of the contributor of the dataset as well as the Distributor in any manuscript or presentation (published or unpublished) using the citation standard documented here: <a  href=\"http://thedata.org/citation\"> http://thedata.org/citation</a> </li> <li> THE DISTRIBUTOR MAKES NO WARRANTIES, EXPRESS OR IMPLIED, BY OPERATION OF LAW OR OTHERWISE, REGARDING OR RELATING TO THE DATASET.   </li> </ul> </div>",
    "specialPermissions": "Submission of the following <a href= \"http://www.murray.harvard.edu/application\"  target=\"_blank\">Application For The Use Of Data</a> is required to access the data from this study.",
    "restrictions": "I will use these data solely for the purposes stated in my application to use data, detailed in a written research proposal. I will honor all agreements and conditions made between the Contributor of the Data and the study participants, and between the Contributor of the Data and the Henry A. Murray Research Archive, Harvard University, as specified in the Memorandum of Agreement.",
    "citationRequirements": "I will include a bibliographic citation acknowledging the use of these data in any publication or presentation in which these data are used. Such citations will appear in footnotes or in the reference section of any such manuscript. I understand the guideline in \"How to Cite This Dataset\" described in the Summary of this study.",
    "depositorRequirements": "Murray Research Archive will list my publication and manuscripts on the Archive website when I submit a bibliographic citation or title of the manuscript, and indicate the Henry A. Murray Research Archive data used. Doing this will also help Henry A. Murray Research Archive to provide funding agencies with essential information about use of archival resources, to fulfill requirements of some memoranda of agreement, and to promote the broader exchange of information about research activities.",
    "conditions": "<b>The following restrictions apply to the use of this data:</b><br />\r\n<b>1. Follow-up and/or contact of participants is not allowed.</b><br />\r\n<b>2. Any request to access video data must be approved by the Director of the Archive, and the significance of the video data for the project must be included in your proposal.</b><br />\r\n<b>3. Any direct or indirect publication on the data must be submitted to the Archive to be shared with the data contributors.</b><br />\r\n<b>4. The data are restricted for use by bonafide researchers with an Institutional Affiliation.</b><br />\r\n",
    "termsOfAccess": "Submission of the following <a href= \"http://www.murray.harvard.edu/application\"  target=\"_blank\">Application For The Use Of Data</a> is required to access the data from this study.<p><b>The following restrictions apply to the use of this data:</b><br /><b>1. Follow-up and/or contact of participants is not allowed.</b><br /><b>2. Any request to access video data must be approved by the Director of the Archive, and the significance of the video data for the project must be included in your proposal.</b><br /><b>3. Any direct or indirect publication on the data must be submitted to the Archive to be shared with the data contributors.</b><br /><b>4. The data are restricted for use by bonafide researchers with an Institutional Affiliation.</b><br /></p>",
    "originalArchive": "Henry A. Murray Research Archive, Institute for Quantitative Social Sciences, Harvard University",
    "availabilityStatus": "Available",
    "contactForAccess": "<a href=\"mailto:mra@help.hmdc.harvard.edu\">Manager of Operations</a>, the Henry A. Murray Research Archive, Institute for Quantitative Social Sciences, 1737 Cambridge St, Cambridge, MA 02138, USA.",

... that's from https://dataverse.harvard.edu/api/datasets/:persistentId/versions/:latest?persistentId=doi%3A10.7910/DVN/BCDSEU

I hope this helps!

Reply:
My feeling is that, since the PR is a 'code cleanup' basically to get R CMD check clean for subsequent development, one should support the change to the license field but leave `termsOfService` for a subsequent PR. I'll update the code to handle both types of license, along the lines of
```diff
diff --git a/R/print.R b/R/print.R
index 564cb71..fa8b582 100644
--- a/R/print.R
+++ b/R/print.R
@@ -54,7 +54,14 @@ print.dataverse_dataset <- function(x, ...) {
             cat("Release Date: ", x$releaseTime, "\n", sep = "")
         }
         if ("license" %in% names(x)) {
-            cat("License: ", x$license$name,"\n", sep = "")
+            if (is.list(x$license)) {
+                ## Dataverse >= 5.10
+                license <- x$license$name
+            } else {
+                ## legacy
+                license <- x$name
+            }
+            cat("License: ", license,"\n", sep = "")
         }
         if ("files" %in% names(x)) {
             n <- NROW(x$files)
```
but this might take a few days for me to find the time to properly test...